### PR TITLE
feat(affiliates): add script to read from google sheet dev mining status

### LINKED ISSUE #2515: Ability to parse through dev mining script to reduce manual reconcilation between excel and dev mining script
Created by David Adams via monday.com integration. 🎉

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David Adams <david@umaproject.org>

**Motivation**

#2515 

**Summary**

Very basic script that reads from google sheets and spits out important data for dev mining. Requires some env vars, documented in code in order to run correctly.

**Details**
This is a low risk script, tested in production against our current google sheet and manually checked between @Tommy1231232 @smb2796 and I for errors. 

image

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #2515

### CODE REVIEW

File: packages/affiliates/apps/GetSheetsDevminingStatus.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+}
+// Validates an eth address and upgrades it to checksum address
+function parseAddress(addr) {
+  assert(web3.utils.isAddress(addr), "Invalid Eth address: " + addr);
+  return web3.utils.toChecksumAddress(addr);
+}
+
+// Assumes dev mining status column === 1 for active emp
+function isEnabled(enabled) {
+  return enabled == "1";
```
Comment (nicholaspai) [inline packages/affiliates/apps/GetSheetsDevminingStatus.js:60]: Consider making this a constant `GOOGLE_EMP_IS_ACTIVE` along with all other hardcoded values that are related to values set in the google sheet

File: packages/affiliates/apps/GetSheetsDevminingStatus.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+// Parses the raw data from sheet
+function parseSheet(sheet) {
+  // Hardcodes columns and names based on the shape of the current dev mining sheet
+  let columns = [
```
Comment (nicholaspai) [inline packages/affiliates/apps/GetSheetsDevminingStatus.js:66]: See above, consider making this a constant mapping

**Review by nicholaspai — APPROVED** [review]
Looks good to me, there's just a lot of hard coded values here so I'd recommend using constant variables to make it more readable (and easier to change in the future)

File: packages/affiliates/apps/GetSheetsDevminingStatus.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    scope: SCOPES
+  });
+  console.log("Authorize this app by visiting this url:", authUrl);
+  prompt.start();
+  const { code } = await prompt.get(["code"]);
+  console.log("got code:", code);
+  return oAuth2Client.getToken(code);
+}
+
+// This assumes URL in teh format https://etherscan.io/address/0xd81028a6fbaaaf604316f330b20d24bfbfd14478
```
Comment (chrismaree) [inline packages/affiliates/apps/GetSheetsDevminingStatus.js:46]:
```suggestion
// This assumes URL in the format https://etherscan.io/address/0xd81028a6fbaaaf604316f330b20d24bfbfd14478
```

File: packages/affiliates/apps/GetSheetsDevminingStatus.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
+// This grabs the latest state from dev mining. Outputs it into a format which can be used in multiple ways.
+// Can be used to generate issue in github, or dev mining configuration, or updating dev mining status github.
+// This Requires ENV vars and a user with access to our private dev mining spreadsheet
+// GOOGLE_CREDENTIALS - set up sheet api https://developers.google.com/sheets/api/quickstart/nodejs
+// SHEET_ID - the sheet ID in the url to the dev mining sheet
+// GOOGLE_TOKEN_PATH - oath2 will need to save a token to your drive. Recommend /tmp/google_token
```
Comment (chrismaree) [inline packages/affiliates/apps/GetSheetsDevminingStatus.js:6]: can you add a comment on how to run the script?

**Review by chrismaree — APPROVED** [review]
Looks pretty awesome! keen to have a set of these scripts to make this process easier. can add one soon for liquidity mining and then add one to create PRs.

**Pull request merged**